portfolio/src/main/webapp/calendar.js

```diff
@@ -49,11 +55,39 @@ function createNewCalendarEvent() {
     document.getElementById('event-end-time-warning').style.visibility =
         'hidden';
     const newCalendarEvent = new CalendarEvent(eventName, startTime, endTime);
-    updateCalendarEventList(newCalendarEvent);
+
+    // Get all the events currently displayed on the UI.
+    // Look through this set of events to ensure that duplicate events
+    // do not get added again.
+    const newEventJson = JSON.stringify(newCalendarEvent);
+    const allEventJson = collectAllEvents();
+
+    let doesEventExist = false;
+    allEventJson.forEach((existingEvenJson) => {
+      if (newEventJson === existingEvenJson) {
+        doesEventExist = true;
+      }
+    });
+
+    if (!doesEventExist) {
+      document.getElementById('event-duplicate-warning').style.visibility =
+          'hidden';
+      updateCalendarEventList(newCalendarEvent);
+    } else {
+      document.getElementById('event-duplicate-warning').style.visibility =
+          'visible';
+    }
     document.getElementById('new-event-name').value = 'New Event';
```

Comment:
Similarly let's see if we can make 'New Event' some constant that does not need to be hard-coded in multiple places.